src/emails/templates/EmailFooter.tsx

```diff
@@ -127,3 +127,117 @@ export const EmailFooter = (props: Props) => {
     </>
   );
 };
+
+export const RedesignedBreachEmailFooter = (props: Props) => {
+  const l10n = props.l10n;
+  const supportLinkUrlObject = new URL(CONST_URL_SUMO_MONITOR_SUPPORT_CENTER);
+  supportLinkUrlObject.searchParams.set("utm_medium", "email");
+  supportLinkUrlObject.searchParams.set("utm_source", "monitor-product");
+  supportLinkUrlObject.searchParams.set("utm_campaign", props.utm_campaign);
+  supportLinkUrlObject.searchParams.set("utm_content", "support-center");
```

Comment:
Ah! Right, I keep misunderstanding how that function works. However, it looks like it also needs a separate `searchParams` object, in which case it wouldn't actually get shorter while adding a layer of indirection:

```suggestion
  const supportLinkUrlObject = new URL(CONST_URL_SUMO_MONITOR_SUPPORT_CENTER);
  supportLinkUrlObject.search = "?" + modifyAttributionsForUrlSearchParams(
    supportLinkUrlObject.searchParams,
    {
      utm_source: "monitor-product",
      utm_campaign: props.utm_campaign,
      utm_content: "support-center",
    },
    {},
  ).toString();
```

I don't think that's more readable or error-proof, but if we already use it elsewhere and people are used to it, then maybe I should just align with that?

Reply:
Either is fine. I don’t think it’s more error-proof, but I don’t dislike the idea of aligning how we are appending `searchParams`. The current function as is was mostly used for modifying query params, but also can be very well be used for just adding them — we might need to consider renaming or splitting the functions a bit. `modifyAttributionsForUrl` would work without taking `searchParams` as an argument.

Reply:
One approach that could make this simpler for callers would be to refactor the `src/app/functions/universal/attributions.ts` helpers to instead extend the `URL` and `URLSearchParams` interfaces, e.g. `export class AttributedURL extends URL` and encapsulate the functionality that the current `modifyAttributionsForUrl` and `modifyAttributionsForUrlSearchParams` helper functions provide into that class.

Then callers could do something like:
```
const supportLinkUrlObject = new AttributedURL(CONST_URL_SUMO_MONITOR_SUPPORT_CENTER);
```

And equivalent for `AttributedURLSearchParams` (although I think if `AttributedURL.searchParams` was `AttributedURLSearchParams` then it would probably not be necessary to export directly, callers could just modify an existing `AttributedURL`'s `searchParams` and get the behavior they want).

That would allow callers to use the same API but it'd be clear from the class name(s) that these are overridden. These are long-stable interfaces so it seems unlikely to break in some subtle way.

Reply:
Ah right, I hadn't noticed that these are two different functions. I suppose I could use `modifyAttributionsForUrl` for a slightly shorter version - I'd assume I'd pass an empty object for the third parameter (`defaultValues`), but I see many calls to that actually use that for UTM parameters. Does either of you know why those calls don't just override them either? I'm just wondering if we can't just simplify it to a `setUrlParams` function - i.e. why do we even need a `defaultValues` parameter?